pilot/pkg/security/authz/model/model.go

```diff
@@ -201,6 +201,10 @@ func generatePermission(rl ruleList, forTCP bool, action rbacpb.RBAC_Action) (*r
 		}
 		and = append(and, ret...)
 	}
+	if action == rbacpb.RBAC_DENY {
```

Comment:
It looks like this will break the current behavior

Reply:
See this warning https://github.com/istio/istio/blob/master/pkg/config/validation/validation.go#L2120-L2125

Reply:
> See this warning https://github.com/istio/istio/blob/master/pkg/config/validation/validation.go#L2120-L2125

Declaring that it only works for a specific port is one way to do this, but obviously this increases the cost of understanding for the user and a better way would be to:

Rules in effect for http only should not be applied to tcp filters (envoy.filters.network.rbac)